keras/backend/mxnet_backend.py

```diff
@@ -5413,7 +5503,8 @@ def _adjust_module(self, inputs, phase):
 
             if not self._module.binded:
                 # allow prediction without compiling the model using different binding
-                if not self.compiled and phase == 'pred':
+                if phase == 'pred' and ((hasattr(self._context[0], 'device_type') and
+                                        self._context[0].device_type == 'eia') or not self.compiled):
```

Comment:
why need to check eia context here? EIA is only used for inference and we don't need to init optimizers.

Reply:
Agreed. This line has no impact because when context is 'eia', the phase is always 'pred' and execution of this logic will depend on 'not self.compiled' which is same as previous condition - 'if not self.compiled and phase == 'pred''

Reply:
Fixed with latest commit.